# Metagenomic analysis workshop Fall 2019
The National Center for Genome Analysis Support (NCGAS) offers a three-day workshop on High Performance Computing (HPC) usage and metagenomic analysis

The National Center for Genome Analysis Support (NCGAS) offers a three-day workshop on High Performance Computing (HPC) usage and metagenomic analysis. 

**Dates** – Oct 7th to 9th, 2019 \
**Location** – Bloomington, IN \
**Registration** – https://redcap.uits.iu.edu/surveys/?s=NYE8DET3EN \
**Deadline to apply** - August 15th, 2019 

# Workshop Information 
The workshop consists of discussions, lectures and hands-on tutorials, to cover topics important to constructing and analyzing metagenomic datasets. Material covers availability of HPC resources, and how to access these resources, while learning how to assemble and annotate metagenomics datasets. The goal of this workshop is to provide a comprehensive preparation for metagenomic analysis and other bioinformatic tasks. 
Metagenomic analysis will consist of learning how to the programs locally, assemble, and annotate the datasets to identify taxa and function. This workshop will walk through the general steps taken in most metagenomics studies. In addition, we will discuss other resources that support further downstream analysis to answer various research questions. While material will make heavy use of XSEDE and IU machines, the material is transferable to any cluster.
Participation in this workshop is by invitation—you must register to be considered. Participation is limited by facility space.

# Prerequisites
This workshop is geared for beginners, but basic Unix commands will not be covered. As such, participants must have basic Linux functionality (sign in, moving around file system, etc.), but expertise is not required. It would be helpful if participants had some exposure to using a cluster for compute jobs and an idea of the end goals for their data.
Beginner tutorials in Linux (bash) can be found here: \
Getting Started on Unix Walkthrough (pdf) (http://pages.iu.edu/~ss93/tutorials/GettingStartedonUnixWalkthrough.pdf)\
Code Academy (https://www.codecademy.com/catalog/subject/all) 

# Objectives
Participants should leave with the following knowledge:\
- Familiarity with nationally available compute resources\
- An understanding of the differences, pros, and cons of VMs, Gateways, Clusters, and Clouds\
- How to run and optimize a job submission on a cluster \
- How to manage large data sets and move data between resources \
- Basic steps in software installation and how to reach out for help when running into errors \
- How to run metagenomics assembly and annotation \
- Ability to reconstruct metagenome assembled genomes and check their quality \
- How to get help for both genomic and computational questions \

Participant data will not be analyzed during the workshop (that takes more time than we have!), but the entire pipeline will be used by participants with smaller scale demo data. We can also schedule online meetings after the workshop to discuss your data. 

# Schedule
## Day 1: 
*8:00 AM to 5:00 PM* \
Introduction to NCGAS staff \
Introduction to clusters and usage\
Optimizing job submission \
Data center tour\
Data management and movement\ 
Overview of metagenomics data collection\

## Day 2: 
*9:00 AM to 5:00 PM* \
Quality control and pre-assessment of the data\
Introduction to metagenome assembly and assembly statistics\
Installing programs locally \
Assembling metagenome assembled genomes and quality assessment\ 
Taxa and function identification\
Downstream analysis\

## Day 3 
*9:00 AM to 1:00 PM*\
Adding more datasets from SRA\
Visualization using R 
 

